src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AReactor.java

```diff
@@ -1,454 +1,475 @@
+			BlockFace.NORTH_EAST,
+			BlockFace.EAST,
+			BlockFace.SOUTH_EAST,
+			BlockFace.SOUTH,
+			BlockFace.SOUTH_WEST,
+			BlockFace.WEST,
+			BlockFace.NORTH_WEST
+		};
+
+	private Set<MachineFuel> recipes = new HashSet<MachineFuel>();
+
+	private static final int[] border = {0, 1, 2, 3, 5, 6, 7, 8, 12, 13, 14, 21, 23};
+	private static final int[] border_1 = {9, 10, 11, 18, 20, 27, 29, 36, 38, 45, 46, 47};
+	private static final int[] border_2 = {15, 16, 17, 24, 26, 33, 35, 42, 44, 51, 52, 53};
+	private static final int[] border_3 = {30, 31, 32, 39, 41, 48, 50};
+	private static final int[] border_4 = {25, 34, 43}; // No coolant border
+	private static final int infoSlot = 49;
+
+	public AReactor(Category category, ItemStack item, String id, RecipeType recipeType, ItemStack[] recipe) {
+		super(category, item, id, recipeType, recipe);
+
+		new BlockMenuPreset(id, getInventoryTitle()) {
+
+			@Override
+			public void init() {
+				constructMenu(this);
+			}
+
+			@Override
+			public void newInstance(final BlockMenu menu, final Block b) {
+				try {
+					if (BlockStorage.getLocationInfo(b.getLocation(), "reactor-mode") == null){
+						BlockStorage.addBlockInfo(b, "reactor-mode", "generator");
+					}
+					if (!BlockStorage.hasBlockInfo(b) || BlockStorage.getLocationInfo(b.getLocation(), "reactor-mode").equals("generator")) {
+						menu.replaceExistingItem(4, new CustomItem(CustomSkull.getItem("eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvOTM0M2NlNThkYTU0Yzc5OTI0YTJjOTMzMWNmYzQxN2ZlOGNjYmJlYTliZTQ1YTdhYzg1ODYwYTZjNzMwIn19fQ=="), "&7Focus: &eElectricity", "", "&6Your Reactor will focus on Power Generation", "&6If your Energy Network doesn't need Power", "&6it will not produce any either", "", "&7> Click to change the Focus to &eProduction"));
+						menu.addMenuClickHandler(4, (p, slot, item, action) -> {
+							BlockStorage.addBlockInfo(b, "reactor-mode", "production");
+							newInstance(menu, b);
+							return false;
+						});
+					}
+					else {
+						menu.replaceExistingItem(4, new CustomItem(SlimefunItems.PLUTONIUM, "&7Focus: &eProduction", "", "&6Your Reactor will focus on producing goods", "&6If your Energy Network doesn't need Power", "&6it will continue to run and simply will", "&6not generate any Power in the mean time", "", "&7> Click to change the Focus to &ePower Generation"));
+						menu.addMenuClickHandler(4, (p, slot, item, action) -> {
+							BlockStorage.addBlockInfo(b, "reactor-mode", "generator");
+							newInstance(menu, b);
+							return false;
+						});
+					}
+					BlockMenu ap = getAccessPort(b.getLocation());
+					if(ap != null) {
+						menu.replaceExistingItem(infoSlot, new CustomItem(new ItemStack(Material.GREEN_WOOL), "&7Access Port", "", "&6Detected", "", "&7> Click to view Access Port"));
+						menu.addMenuClickHandler(infoSlot, (p, slot, item, action) -> {
+						ap.open(p);
+						newInstance(menu, b);
+							
+							return false;
+						});

```

Comment:
Please fix the formatting on the above lines.

Reply:
Should be fixed now, reformatted to correct indentation